source/game/kart/KartHalfPipe.cc

```diff
@@ -37,53 +39,55 @@ void KartHalfPipe::calc() {
 
     calcTrick();
 
-    if (m_touchingZipper && state()->isAirStart()) {
-        dynamics()->setExtVel(EGG::Vector3f::zero);
-        state()->setOverZipper(true);
+    if (collide()->surfaceFlags().offBit(KartCollide::eSurfaceFlags::StopHalfPipeState)) {
+        if ((m_touchingZipper && state()->isAirStart())) {
```

Comment:
```suggestion
    if (collide()->surfaceFlags().offBit(KartCollide::eSurfaceFlags::StopHalfPipeState) && m_touchingZipper && state()->isAirStart()) {
```
Needs formatting, but avoids a lot of diff noise.